charts/mimir-distributed/templates/store-gateway/store-gateway-statefulset.yaml

```diff
@@ -101,8 +112,12 @@ spec:
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           args:
             - "-target=store-gateway"
-            - -activity-tracker.filepath=/active-query-tracker/activity.log
+            - "-activity-tracker.filepath=/active-query-tracker/activity.log"
             - "-config.file=/etc/mimir/mimir.yaml"
+            {{- if .Values.store_gateway.rollout_operator.enabled }}
+            - "-store-gateway.sharding-ring.instance-availability-zone={{ $rolloutZone.name }}"
+            - "-store-gateway.sharding-ring.zone-awareness-enabled=true"
```

Comment:
We're trying to have everything possible defined in `mimir.config` to avoid surprises for the customer during upgrades (suddenly there's a new cli flag being used). Obviously `instance-availability-zone` is going to be an exception, but zone-awarness-enabled should go there in theory - I have to think about this one.

Reply:
Let move `zone-awarness-enabled` to mimir.config and I'll add a validation in validation.yaml that checks for consistency